aws_lambda_powertools/utilities/batch/base.py

```diff
@@ -100,6 +102,40 @@ def process(self) -> List[Tuple]:
         """
         return [self._process_record(record) for record in self.records]
 
+    @abstractmethod
+    async def _async_process_record(self, record: dict):
+        """
+        Async process record with handler.
+        """
+        raise NotImplementedError()
+
+    def async_process(self) -> List[Tuple]:
+        """
+        Async call instance's handler for each record.
+        """
+
+        async def async_process():
+            return list(await asyncio.gather(*[self._async_process_record(record) for record in self.records]))
+
+        # WARNING
+        # Do not use "asyncio.run(async_process())" due to Lambda container thaws/freeze, otherwise we might get "Event Loop is closed"
+        # Instead, get_event_loop() can also create one if a previous was erroneously closed
+        # More: https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtime-environment.html#runtimes-lifecycle-shutdown
+        # Extra: just follow how the well-tested mangum library do:
+        #       https://github.com/jordaneremieff/mangum/discussions/256#discussioncomment-2638946
+        #       https://github.com/jordaneremieff/mangum/blob/b85cd4a97f8ddd56094ccc540ca7156c76081745/mangum/protocols/http.py#L44
+
+        # Detect environment and create a loop for each one
+        coro = async_process()
+        if os.environ.get('AWS_LAMBDA_RUNTIME_API'):
+            # Running in lambda server
+            loop = asyncio.get_event_loop()
+            task_instance = loop.create_task(coro)
+            return loop.run_until_complete(task_instance)
+        else:
+            # Running in another place like local tests
+            return asyncio.run(coro)
```

Comment:
@heitorlessa please check this, I get from Mangum that have more experience with async in lambdas

Reply:
I need to read more on this. `AWS_LAMBDA_RUNTIME_API` might not be the best option.

Reply:
updated to TASK_ROOT as it's guaranteed to be in proper emulators too.